### ISSUE TITLE
WebDriver 3.3.1 and Safari Technology Preview 10.2 raises NoSuchWindowException

### ISSUE THREAD

**Issue by MentalDisruptor** [SeleniumHQ/selenium#3854]
## Meta -
OS: Mac OS 10.12.4
Selenium Version: Selenium Server 3.3.1 (WebDriver 3.3.1)
Browser: Safari Technology Preview

Browser Version:  
Release 27 (Safari 10.2, WebKit 12604.1.15.2)
Remote Automation enabled:
![safari_tp_automation_settings](https://cloud.githubusercontent.com/assets/1024384/25129084/cd550188-243c-11e7-89a0-07a8683e1cf2.png)

## Expected Behavior -
Window opens and test automation is able to execute given commands.

## Actual Behavior -
Window opens and immediately closes again. Exception is raised:
org.openqa.selenium.NoSuchWindowException: A request to use a window could not be satisfied because the window could not be found.
(See attached [stacktrace.txt](https://github.com/SeleniumHQ/selenium/files/929076/stacktrace.txt))

## Steps to reproduce -

Java-Codesnippet:
```
                DesiredCapabilities desiredCaps = DesiredCapabilities.safari();

                SafariOptions sOptions = new SafariOptions();
                sOptions.setUseCleanSession(true); // init a clean Safari session at all times
                sOptions.setUseTechnologyPreview(true); // enable Technology Preview

                desiredCaps.setCapability(SafariOptions.CAPABILITY, sOptions);
                WebDriver driver = new RemoteWebDriver(
                        new URL("http://url.to.seleniumgrid.hub:4444/wd/hub"),
                        desiredCaps
                );
                driver.open("http://www.example.org");
```

**Comment by pnaranja** [comment]
FYI, I got the same exception when trying Safari Tech Preview Release 28.
I also tried Selenium 3.4.0 with Safari tech preview

OS: Mac OS 10.12.4
Selenium Version: Selenium Server 3.3.1 (WebDriver 3.3.1) (Java)
Selenium Version: Selenium Server 3.4.0 (WebDriver 3.4.0) (Java)
Browser: Safari Technology Preview Release 28

**Comment by mike123vn** [comment]
Any solution right now?

**Comment by webtoman** [comment]
I get this with Safari Version 10.1 (12603.1.30.0.34) as well. Tested with selenium server 3.0.0, 3.0.1, 3.3.1 and 3.4.0, all fail the same.

**Comment by sky-joaonabais** [comment]
Hello,

Any news on this?

I'm also getting this, when I try to set Window size (at the very beginning).

Selenium:  3.4.0
Mac OS X: 10.12.6
Safari:        10.1.2 (12603.3.8)

Log:
```
12:09:48.385 INFO - Done: [implicitly wait: 10000]
12:09:48.399 INFO - Executing: [set window size])
12:09:48.457 WARN - Exception thrown
org.openqa.selenium.NoSuchWindowException: A request to use a window could not be satisfied because the window could not be found. (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 57 milliseconds
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'OSTML0231871', ip: '10.241.11.98', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.6', java.version: '1.8.0_101'
Driver info: org.openqa.selenium.safari.SafariDriver
Capabilities [{applicationCacheEnabled=true, rotatable=false, databaseEnabled=true, handlesAlerts=true, version=12603.3.8, cleanSession=true, platform=MAC, nativeEvents=true, locationContextEnabled=false, webStorageEnabled=true, browserName=safari, javascriptEnabled=true, platformName=macOS, cssSelectorsEnabled=true}]
Session ID: 87B47DAC-867D-4CDD-85BC-25B15A6A95F1
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:215)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:167)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:671)
	at org.openqa.selenium.remote.RemoteWebDriver$RemoteWebDriverOptions$RemoteWindow.setSize(RemoteWebDriver.java:890)
	at org.openqa.selenium.support.events.EventFiringWebDriver$EventFiringWindow.setSize(EventFiringWebDriver.java:625)
	at org.openqa.selenium.remote.server.handler.SetWindowSize.call(SetWindowSize.java:53)
	at org.openqa.selenium.remote.server.handler.SetWindowSize.call(SetWindowSize.java:27)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
12:09:48.457 WARN - Exception: A request to use a window could not be satisfied because the window could not be found. (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 57 milliseconds
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'```

**Comment by xmo-odoo** [comment]
According to Nikol Marku in https://groups.google.com/forum/#!topic/selenium-users/xEGcK92rzVg

> UnsupportedCommandException for is Displayed NOT FIXED and maximize does not work and surely there will lot more 

so this is likely an issue with the Safari side not implementing (or rejecting) these commands, and should probably be reported to Webkit/Safari/Apple rather than Selenium?

**Comment by FA-tvonmoll** [comment]
FWIW: Window sizing has been working for me with the newly-released Safari 11.0.

**Comment by barancev** [comment]
This issue can't be fixed on Selenium side, it should be fixed in safaridriver. Please report it to the Apple's bug tracker if it is still actual.
